apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts

```diff
@@ -836,11 +836,11 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
         this.invoiceConfirmationModel.show();
     }
 
-    public deleteConfirmedInvoice() {
+    public deleteConfirmedInvoice(voucherUniqueName?: any) {
         this.invoiceConfirmationModel.hide();
         if (this.selectedVoucher === VoucherTypeEnum.purchase) {
             const requestObject = {
-                uniqueName: (this.selectedInvoice) ? this.selectedInvoice.uniqueName : (this.selectedInvoiceForDetails) ? this.selectedInvoiceForDetails.uniqueName : ''
+                uniqueName: (voucherUniqueName) ? voucherUniqueName : (this.selectedInvoice) ? this.selectedInvoice.uniqueName : (this.selectedInvoiceForDetails) ? this.selectedInvoiceForDetails.uniqueName : ''
```

Comment:
Consider adding `encodeURIComponent` in `voucherUniqueName`, `this.selectedInvoice.uniqueName` & `this.selectedInvoiceForDetails.uniqueName`